mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java

```diff
@@ -120,43 +135,51 @@ public DaoDeleteMethodGenerator(
                     + "instance must have an 'entityClass' argument",
                 Delete.class.getSimpleName());
         return Optional.empty();
+      } else {
+        entityDefinition = context.getEntityFactory().getDefinition(entityElement);
       }
-      entityDefinition = context.getEntityFactory().getDefinition(entityElement);
-      // if a custom if clause is provided, the whole primary key must also be provided.
-      List<? extends VariableElement> primaryKeyParameters = parameters;
-      if (!annotation.customIfClause().isEmpty()) {
-        if (primaryKeyParameters.size() < entityDefinition.getPrimaryKey().size()) {
-          List<TypeName> primaryKeyTypes =
-              entityDefinition.getPrimaryKey().stream()
-                  .map(d -> d.getType().asTypeName())
-                  .collect(Collectors.toList());
-          context
-              .getMessager()
-              .error(
-                  methodElement,
-                  "Invalid parameter list: %s methods that have a custom if clause"
-                      + "must specify the entire primary key (expected primary keys of %s: %s)",
-                  Delete.class.getSimpleName(),
-                  entityElement.getSimpleName(),
-                  primaryKeyTypes);
-          return Optional.empty();
-        } else {
-          // restrict parameters to primary key length.
-          primaryKeyParameters =
-              primaryKeyParameters.subList(0, entityDefinition.getPrimaryKey().size());
+
+      if (customWhereClause.isEmpty()) {
+        /* if a custom if clause is provided, the whole primary key must also be provided.
+         * we only do this check if there is no custom where clause as the order of
+         * keys may differ in that case.*/
+        List<? extends VariableElement> primaryKeyParameters = parameters;
+        if (!customIfClause.isEmpty()) {
+          if (primaryKeyParameters.size() < entityDefinition.getPrimaryKey().size()) {
+            List<TypeName> primaryKeyTypes =
+                entityDefinition.getPrimaryKey().stream()
+                    .map(d -> d.getType().asTypeName())
+                    .collect(Collectors.toList());
+            context
+                .getMessager()
+                .error(
+                    methodElement,
+                    "Invalid parameter list: %s methods that have a custom if clause"
+                        + "must specify the entire primary key (expected primary keys of %s: %s)",
+                    Delete.class.getSimpleName(),
+                    entityElement.getSimpleName(),
+                    primaryKeyTypes);
+            return Optional.empty();
+          } else {
+            // restrict parameters to primary key length.
+            primaryKeyParameters =
+                primaryKeyParameters.subList(0, entityDefinition.getPrimaryKey().size());
+          }
```

Comment:
No advanced validation is done on parameters if `customWhereClause` is provided.  We could technically look at the # of parameters and compare to partition key size (since that's required), but not the types since those could be in any order.  I figured adding size validation would be validation for the sake for validation.

Reply:
Agreed, no need to go into details if we can't fully validate anyway.